src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc

```diff
@@ -0,0 +1,311 @@
+    }
+  } else {
+    // The permissions of passenger's device  were checked by GetPermission.
+    // This request can contain only one module
+    // so we don't have to exclude any module.
+    DCHECK(params->get(message_params::kModuleTypes,
+                       Json::Value(Json::arrayValue)).size() == 1);
+  }
+}
+
+void GetInteriorVehicleDataCapabiliesRequest::OnEvent(
+    const can_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (functional_modules::hmi_api::get_interior_vehicle_data_capabilities ==
+      event.id()) {
+    std::string result_code;
+    std::string info;
+
+    Json::Value value;
+    Json::Reader reader;
+    reader.parse(event.event_message()->json_message(), value);
+
+    bool success = ParseResultCode(value, result_code, info);
+
+    validators::ValidationResult validation_result = validators::SUCCESS;
+
+    const int capabilities_min_size = 1;
+    const int capabilities_max_size = 1000;
+
+    if (success) {
+      validators::GetInteriorVehicleDataCapabilitiesResponseValidator validator;
+      if (IsMember(value[kResult], kInteriorVehicleDataCapabilities)) {
+        int capabilities_size =
+            value[kResult][kInteriorVehicleDataCapabilities].size();
+        if (value[kResult][kInteriorVehicleDataCapabilities].isArray() &&
+            (capabilities_size >= capabilities_min_size) &&
+            (capabilities_size <= capabilities_max_size)) {
+          for (int i = 0; i < capabilities_size; ++i) {
+            validator.Validate(
+                value[kResult][kInteriorVehicleDataCapabilities][i],
+                response_params_[kInteriorVehicleDataCapabilities][i]);
+          }
+        } else {
+          validation_result = validators::INVALID_DATA;
+        }
+      } else {
+        validation_result = validators::INVALID_DATA;
+      }
+
+      if (validators::SUCCESS != validation_result) {
+        success = false;
+        info = "Invalid response from the vehicle.";
+        result_code = result_codes::kGenericError;
+      }
+    }
+
+    if (!success) {
+      //  Try to read capabilities from file.
+      if (ReadCapabilitiesFromFile()) {
+        success = true;
+        result_code = result_codes::kSuccess;
+        info = "";
+      }
+    }
+
+    SendResponse(success, result_code.c_str(), info);
+  } else {
+    LOG4CXX_ERROR(logger_, "Received unknown event: " << event.id());
+  }
+}
+
+bool GetInteriorVehicleDataCapabiliesRequest::ReadCapabilitiesFromFile() {
+  LOG4CXX_INFO(logger_,
+               "Failed to get correct response from HMI; \
+      trying to read from file");
+  Json::Value request;
+  Json::Reader reader;
+  if (!reader.parse(message_->json_message(), request)) {
+    LOG4CXX_ERROR(logger_, "Failed to read capabilities from file also");
+    return false;
+  }
+
+  VehicleCapabilities file_caps;
+  Json::Value zone_capabilities;
```

Comment:
@AByzhynar 
Currently there is no clarification from customer on format expected for fallback capabilities, so I would leave these parts untill further information will be provided and than change that accordingly.